components/AposTemplate.astro

```diff
@@ -31,7 +31,7 @@ if (aposData.errorFetchingPage || aposData.page?.type === 'apos-fetch-error') {
 
 
 const Template = templates[aposData.template]
-  || templates[aposData.template.replace(':page', '')]
+  || templates[aposData.template?.replace(':page', '')]
```

Comment:
This would break the template if no template is available (e.g. sitemap) and makes the checks after pointless.